### PR TITLE
Update conjur-env go.mod dependencies

### PR DESCRIPTION
**Pull request by andytinkham**
Signed-off-by: Andy Tinkham <andy.tinkham@cyberark.com>

### Desired Outcome

Update dependencies to latest versions in conjur-env/go.mod

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [X] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [X] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [X] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [X] No behavior was changed with this PR

#### Security

- [X] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [2.2.3] - 2022-06-07
 
 ### Changed
+- Updated conjur-env dependencies to latest versions (github.com/cyberark/summon -> v0.9.2,
```
Comment (john-odonnell) [inline CHANGELOG.md:12]: We just cut the 2.2.3 release - should this be in either the Unreleased or a new 2.2.4 section?

**Review by john-odonnell — COMMENTED** [review]
Changelog update might need to be in a new section, but LGTM otherwise 👍

File: CHANGELOG.md (hunk 1 above)
Comment (andytinkham) [reply CHANGELOG.md:12]: Yes it should, and I could have sworn I checked that. Will update. Thanks!

**Review by john-odonnell — APPROVED** [review]
LGTM 👍

**Pull request merged**